File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        ArrayList<String> stack = new ArrayList<>();
        for(String token : tokens){
            if (token.equals("+")){
                Integer op1 = Integer.parseInt(stack.remove(stack.size() - 1));
                Integer op2 = Integer.parseInt(stack.remove(stack.size() - 1));
                stack.add(Integer.toString(op1 + op2));
            } else if (token.equals("-")){
                Integer op1 = Integer.parseInt(stack.remove(stack.size() - 1));
                Integer op2 = Integer.parseInt(stack.remove(stack.size() - 1));
                stack.add(Integer.toString(op2 - op1));
            } else if (token.equals("*")){
                Integer op1 = Integer.parseInt(stack.remove(stack.size() - 1));
                Integer op2 = Integer.parseInt(stack.remove(stack.size() - 1));
                stack.add(Integer.toString(op1 * op2));
            } else if (token.equals("/")){
                Integer op1 = Integer.parseInt(stack.remove(stack.size() - 1));
                Integer op2 = Integer.parseInt(stack.remove(stack.size() - 1));
                stack.add(Integer.toString(op2 / op1));
            } else {
                stack.add(token);
            }
        }
        
        return Integer.parseInt(stack.remove(stack.size() - 1));
    }
}

File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        specials = {'+','-','*','/'}
        
        stack = []
        for token in tokens:
            if token not in specials:
                stack.append(int(token))
            else:
                b, a = stack.pop(), stack.pop()
                if token == '+':
                    stack.append(a + b)
                elif token == '-':
                    stack.append(a - b)
                elif token == '*':
                    stack.append(a * b)
                elif token == '/':
                    stack.append(int(a / b))
        return stack[-1]